adapters/matterfull/matterfull.go

```diff
@@ -0,0 +1,221 @@
+		if err != nil {
+			errs = append(errs, err)
+			return nil, errs
+		}
+		result = append(result, bidRequest)
+	}
+	return result, errs
+}
+
+// getImpressionsInfo checks each impression for validity and returns impressions copy with corresponding exts
+func getImpressionsInfo(imps []openrtb2.Imp) (map[openrtb_ext.ExtImpMatterfull][]openrtb2.Imp, []openrtb2.Imp, []error) {
+	var errors []error
+	resImps := make([]openrtb2.Imp, 0, len(imps))
+	res := make(map[openrtb_ext.ExtImpMatterfull][]openrtb2.Imp)
+
+	for imp := range iterutil.SlicePointerValues(imps) {
+		impExt, err := getImpressionExt(imp)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+		//dispatchImpressions
+		//Group impressions by Matterfull-specific parameters `pid
+		if err := compatImpression(imp); err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		res[impExt] = append(res[impExt], *imp)
+		resImps = append(resImps, *imp)
+	}
+	return res, resImps, errors
+}
+
+// Alter impression info to comply with Matterfull platform requirements
+func compatImpression(imp *openrtb2.Imp) error {
+	imp.Ext = nil //do not forward ext to Matterfull platform
+	if imp.Banner != nil {
+		return compatBannerImpression(imp)
+	}
+	return nil
+}
+
+func compatBannerImpression(imp *openrtb2.Imp) error {
+	// Create a copy of the banner, since imp is a shallow copy of the original.
+
+	bannerCopy := *imp.Banner
+	banner := &bannerCopy
+	//As banner.w/h are required fields for Matterfull platform - take the first format entry
+	if banner.W == nil || banner.H == nil {
+		if len(banner.Format) == 0 {
+			return &errortypes.BadInput{Message: "Expected at least one banner.format entry or explicit w/h"}
+		}
+		format := banner.Format[0]
+		banner.Format = banner.Format[1:]
+		banner.W = &format.W
+		banner.H = &format.H
+		imp.Banner = banner
+	}
+	return nil
+}
+
+func getImpressionExt(imp *openrtb2.Imp) (openrtb_ext.ExtImpMatterfull, error) {
+	var bidderExt adapters.ExtImpBidder
+	if err := jsonutil.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		return openrtb_ext.ExtImpMatterfull{}, &errortypes.BadInput{
+			Message: err.Error(),
+		}
+	}
+	var MatterfullExt openrtb_ext.ExtImpMatterfull
+	if err := jsonutil.Unmarshal(bidderExt.Bidder, &MatterfullExt); err != nil {
+		return openrtb_ext.ExtImpMatterfull{}, &errortypes.BadInput{
+			Message: err.Error(),
+		}
+	}
+	return MatterfullExt, nil
+}
+
+func (adapter *adapter) buildAdapterRequest(prebidBidRequest *openrtb2.BidRequest, params *openrtb_ext.ExtImpMatterfull, imps []openrtb2.Imp) (*adapters.RequestData, error) {
+	newBidRequest := createBidRequest(prebidBidRequest, params, imps)
+	reqJSON, err := jsonutil.Marshal(newBidRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("x-openrtb-version", "2.5")
```

Comment:
What your bidder requires is in your hands, but I note that your yaml says you support 2.6, and prebid uses 2.6 internally, so what you are sending would actually be 2.6. Not going to hold up approval because you are telling yourself that this is 2.5, but want to make sure this is an intentional choice rather than a typo.